runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/streaming/ComputationState.java

```diff
@@ -74,8 +79,22 @@ public ImmutableMap<String, String> getTransformUserNameToStateFamily() {
     return transformUserNameToStateFamily;
   }
 
-  public ConcurrentLinkedQueue<ExecutionState> getExecutionStateQueue() {
-    return executionStateQueue;
+  /**
+   * Adds the {@link ExecutionState} to the internal {@link #executionStateQueue} so that it can be
+   * re-used in future processing.
+   */
+  public void releaseExecutionState(ExecutionState executionState) {
+    executionStateQueue.offer(executionState);
```

Comment:
how do we reset the execution state? from my understanding in the code, the executionState gets closed when work processing fails due to some exception

in the current code we set the local reference to executionState to null since there is a bit of processing we do after in `StreamingDataflowWorker.process()`, but in the new code the local reference will be released once we exit the method scope of `StreamingWorkScheduler.executeWork()`